gcpvault_test.go

```diff
@@ -63,6 +64,7 @@ func TestGetSecrets(t *testing.T) {
 			givenCfg: Config{
 				Role:       "my-gcp-role",
 				SecretPath: "my-secret-path",
+				MaxRetries: 1,
```

Comment:
shall we test with more retries?

Reply:
I think I don't see where there's a check that 1) multiple requests happen and 2) that multiple requests happen and it eventually succeeds and 3) that multiple requests happen and it eventually fails. would be good to add that.